### DISCUSSION TITLE
How to update external config files with sdkman candidate paths

### DISCUSSION THREAD

**Discussion by tsaarni** [sdkman/sdkman-cli#1103] [Q&A]
Some applications require JDK paths to be cofigured. For example, in vscode and vscode-java extension, following snipped could be part of user's settings `~/.config/Code/User/settings.json`.

```json
    "java.configuration.runtimes": [
        {
            "name": "JavaSE-17",
            "path": "/home/tsaarni/.sdkman/candidates/java/current"
        },
        {
            "name": "JavaSE-11",
            "path": "/home/tsaarni/.sdkman/candidates/java/11.0.15-tem/"
        },
        {
            "name": "JavaSE-1.8",
            "path": "/home/tsaarni/.sdkman/candidates/java/8.0.332-tem/"
        }
    ],
```

After I update the packages with sdkman, the paths will become invalid since they contain the patch level version.

Are there recommended approaches for keeping external config files in sync?

Maybe one approach could be to have stable symlinks like `~/.sdkman/candidates/java/[major.minor]-tem` that would be automatically updated to point to the latest `[major.minor.patch]-tem` candidate. Then it would be unnecessary to change the external configs at each JDK update. Can this be achieved currently?

**Comment by mims-github** [comment]
Or even just the major version, if one is interested to always use the latest of a given major version.
